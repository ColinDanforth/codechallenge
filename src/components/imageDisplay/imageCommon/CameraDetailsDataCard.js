import React from 'react'
import LensDetailsObject from "./LensDetailsObject"

const userCardBackground={
  marginTop: '2%',
  backgroundColor: 'RGBA(256,256,256,1)',
  borderBottom:'inset',
}

const textStyle={
  marginTop: '1%',
  marginLeft: '2%',
  marginRight: '2%',
  marginBottom: '1%',
}

const isThereLensInfo = (data) => {
  if(data.hasOwnProperty('lens_info') && typeof data.lens_info === 'object' && data.lens_info.hasOwnProperty('friendly_name')> 0){
    return <LensDetailsObject lensInfo={data.lens_info}/>
  } else if(data.hasOwnProperty('lens') && data.lens !== null && data.lens.length > 0){
    return <p style={textStyle}>Lens: {data.lens}</p>
  }else{
    return <p style={textStyle}>Lens Info: No details Submitted</p>
  }
}

const cameraName = (data) => {
  if (data.hasOwnProperty('camera') && data.camera !== null && data.camera.length > 0) {
    return <p style={textStyle}>Camera: {data.camera}</p>
  }
}
const shutterSpeed = (data) => {
  if (data.hasOwnProperty('shutter_speed') && data.shutter_speed !== null && data.shutter_speed.length > 0) {
    return <p style={textStyle}>Shutter Speed: {data.shutter_speed}</p>
  }
}

const aperture = (data) => {
  if(data.hasOwnProperty('aperture') && data.aperture !== null && data.aperture.length > 0){
    return <p style={textStyle}>Aperture: {data.aperture}</p>
  }
}

const CameraDetailDataCard = (props) => {
  return(
    <div style={userCardBackground}>
      {cameraName(props.dataObject)}
      {shutterSpeed(props.dataObject)}
      {aperture(props.dataObject)}
      {isThereLensInfo(props.dataObject)}
    </div>
  )
}

export default CameraDetailDataCard
