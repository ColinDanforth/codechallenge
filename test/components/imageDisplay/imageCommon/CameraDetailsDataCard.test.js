import React from 'react';
import CameraDetailsDataCard from '../../../../src/components/imageCommon/CameraDetailsDataCard';
import renderer from 'react-test-renderer';

test('CameraDetailsDataCard should display LensDetailCard, and appropriate strings', () => {
  const testDataObject={
    lens_info: {
      friendly_name: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
      features: "Micro Four Thirds System<br/>24-80mm (35mm Equivalent)<br/>Aperture Range: f/2.8 to f/22",
    },
    camera: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
    shutter_speed: 'test string',
    aperture: 'test string',
  }

  const component = renderer.create(
    <CameraDetailsDataCard dataObject={testDataObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('CameraDetailsDataCard should display appropriate strings including lens in place of lens info, excluding camera, ' +
  'shutter speed, and aperture', () => {
  const testDataObject={
    lens: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
    camera: null,
    shutter_speed: null,
    aperture: null
  }

  const component = renderer.create(
    <CameraDetailsDataCard dataObject={testDataObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('CameraDetailsDataCard should display LensDetailCard, and appropriate strings excluding camera, shutter speed, and ' +
  'aperture', () => {
  const testDataObject={
    lens_info: {
      friendly_name: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
      features: "Micro Four Thirds System<br/>24-80mm (35mm Equivalent)<br/>Aperture Range: f/2.8 to f/22",
    },
    camera: null,
    shutter_speed: null,
    aperture: null
  }

  const component = renderer.create(
    <CameraDetailsDataCard dataObject={testDataObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('CameraDetailsDataCard should display "Lens Info: no details"', () => {
  const testDataObject={
    lens: null,
    camera: null,
    shutter_speed: null,
    aperture: null
  }

  const component = renderer.create(
    <CameraDetailsDataCard dataObject={testDataObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
