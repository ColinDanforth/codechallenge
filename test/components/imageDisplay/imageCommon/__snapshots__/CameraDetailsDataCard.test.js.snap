// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraDetailsDataCard should display "Lens Info: no details" 1`] = `
<div
  style={
    Object {
      "backgroundColor": "RGBA(256,256,256,1)",
      "borderBottom": "inset",
      "marginTop": "2%",
    }
  }
>
  <p
    style={
      Object {
        "marginBottom": "1%",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    Lens Info: No details Submitted
  </p>
</div>
`;

exports[`CameraDetailsDataCard should display LensDetailCard, and appropriate strings 1`] = `
<div
  style={
    Object {
      "backgroundColor": "RGBA(256,256,256,1)",
      "borderBottom": "inset",
      "marginTop": "2%",
    }
  }
>
  <p
    style={
      Object {
        "marginBottom": "1%",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    Camera: 
    Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
  </p>
  <p
    style={
      Object {
        "marginBottom": "1%",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    Shutter Speed: 
    test string
  </p>
  <p
    style={
      Object {
        "marginBottom": "1%",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    Aperture: 
    test string
  </p>
  <div>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "30px",
          "marginRight": "30px",
        }
      }
    >
      Lens Info:
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Name: 
      Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Feature: 
      Micro Four Thirds System
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      24-80mm (35mm Equivalent)
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Aperture Range: f/2.8 to f/22
    </p>
  </div>
</div>
`;

exports[`CameraDetailsDataCard should display LensDetailCard, and appropriate strings excluding camera, shutter speed, and aperture 1`] = `
<div
  style={
    Object {
      "backgroundColor": "RGBA(256,256,256,1)",
      "borderBottom": "inset",
      "marginTop": "2%",
    }
  }
>
  <div>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "30px",
          "marginRight": "30px",
        }
      }
    >
      Lens Info:
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Name: 
      Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Feature: 
      Micro Four Thirds System
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      24-80mm (35mm Equivalent)
    </p>
    <p
      style={
        Object {
          "marginBottom": "15px",
          "marginLeft": "60px",
          "marginRight": "30px",
        }
      }
    >
      Aperture Range: f/2.8 to f/22
    </p>
  </div>
</div>
`;

exports[`CameraDetailsDataCard should display appropriate strings including lens in place of lens info, excluding camera, shutter speed, and aperture 1`] = `
<div
  style={
    Object {
      "backgroundColor": "RGBA(256,256,256,1)",
      "borderBottom": "inset",
      "marginTop": "2%",
    }
  }
>
  <p
    style={
      Object {
        "marginBottom": "1%",
        "marginLeft": "2%",
        "marginRight": "2%",
        "marginTop": "1%",
      }
    }
  >
    Lens: 
    Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
  </p>
</div>
`;
